clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -448,6 +448,12 @@ class CIRScopeOpLowering
     mlir::OpBuilder::InsertionGuard guard(rewriter);
     auto loc = scopeOp.getLoc();
 
+    // Empty scope: just remove it.
+    if (scopeOp.getRegion().empty()) {
```

Comment:
I'm fine having this here because it ensures it happens, but I'd expect this to be handled in MergeCleanups pass (perhaps we need a better name for that pass, but this is another discussion), can you add this there too? Can be in  follow up work.

Reply:
@bcardosolopes shouldn't the conversion and the passes be independent of each other?

The way I see it, the conversion should work with or without the cleanup pass.

Reply:
Yes, which is what I mentioned about being fine with it, cause it makes sense. I'd prefer if you could also remove the scope as part of MergeCleanups as well, so we make sure to cleanup those silly little things at the CIR level as well.

Reply:
Ohh, gotcha. I'll add it to the pass.